src/ssd/ssd-extents.c

```diff
@@ -232,18 +220,12 @@ ssd_extents_update(struct ssd *ssd)
 			wlr_scene_rect_set_size(rect, result_box.width,
 				result_box.height);
 			wlr_scene_node_set_position(part->node,
-				target->x + (result_box.x - part_box.x),
-				target->y + (result_box.y - part_box.y));
-			continue;
-		}
-
-		/* Fully visible */
-		if (target->x != part->node->x
-				|| target->y != part->node->y) {
-			wlr_scene_node_set_position(part->node, target->x, target->y);
-		}
-		if (target->width != rect->width || target->height != rect->height) {
-			wlr_scene_rect_set_size(rect, target->width, target->height);
+				target.x + (result_box.x - part_box.x),
+				target.y + (result_box.y - part_box.y));
+		} else {
+			/* Fully visible */
+			wlr_scene_node_set_position(part->node, target.x, target.y);
+			wlr_scene_rect_set_size(rect, target.width, target.height);
```

Comment:
I removed checks like `target->x != part->node->x` as it's done in wlroots side.